core/domain/stats_domain.py

```diff
@@ -123,23 +145,117 @@ def validate(self):
             if not isinstance(self.interaction_id, basestring):
                 raise utils.ValidationError(
                     'Expected interaction_id to be a string, received %s' %
-                    str(self.interaction_id))
+                    self.interaction_id)
 
-            # check if interaction_id is valid
+            # Verify interaction_id is valid.
             if (self.interaction_id not in
                     interaction_registry.Registry.get_all_interaction_ids()):
                 raise utils.ValidationError(
                     'Unknown interaction id %s' % self.interaction_id)
 
-        if not isinstance(self.answers_list, list):
+        if not isinstance(self.submitted_answer_list, list):
             raise utils.ValidationError(
-                'Expected answers_list to be a list, received %s' %
-                self.answers_list)
+                'Expected submitted_answer_list to be a list, received %s' %
+                self.submitted_answer_list)
+
+        for submitted_answer in self.submitted_answer_list:
+            submitted_answer.validate()
+
+
+class SubmittedAnswer(object):
+    """Domain object representing an answer submitted to a state."""
+
+    # There is a danger of data overflow if the answer log exceeds 1 MB. Given
+    # 1000-5000 answers, each answer must be at most 200-1000 bytes in size. We
+    # will address this later if it happens regularly. At the moment, a
+    # ValidationError is raised if an answer exceeds the maximum size.
+    _MAX_BYTES_PER_ANSWER = 2500
+
+    # Prefix of strings that are cropped because they are too long.
+    _CROPPED_PREFIX_STRING = 'CROPPED: '
+
+    # Answer value that is stored if non-string answer is too big
+    _PLACEHOLDER_FOR_TOO_LARGE_NONSTRING = 'TOO LARGE NONSTRING'
+
+    # NOTE TO DEVELOPERS: do not use the rule_spec_str and answer_str
```

Comment:
This comment is cryptic. I don't actually understand it: what does "as they are here as part of the answer migration" mean? Might be worth elaborating.